### PR TITLE
Add a flag to turn off secp256k1 support

### PR DESCRIPTION
**Pull request by michaelpj**
Do so in the windows CI. This should hopefully unbreak people
downstream until we get a better solution.

(Draft until I see the Windows CI pass)

### CODE REVIEW

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -95,12 +95,10 @@ jobs:
...
       run: |
-        mkdir secp256k1-sources
-        cd secp256k1-sources
-        git clone https://github.com/bitcoin-core/secp256k1.git
-        cd secp256k1
-        git reset --hard $SECP256K1_REF
-        echo "FIXME: install from sources"
+        echo "FIXME: can't install from sources, so instead libsecp256k1 support is disabled"
+        cat >> cabal.project <<EOF
+        package cardano-crypto-class
```
Comment (lehins) [inline .github/workflows/haskell.yml:100]: I think it is better to do `cabal configure --flags="-secp256k1-support"'` instead of editing cabal.project

Comment (michaelpj) [reply .github/workflows/haskell.yml:100]: This is what is done further down for configuring libsodium, so I just copied that :shrug:

Comment (michaelpj) [reply .github/workflows/haskell.yml:100]: I also don't think that multiple calls to `cabal configure` will compose in the way we want.

**Comment by michaelpj** [comment]
Oh, it won't pick up my changes to the workflow until this is merged, annoyingly, so we may just have to try it and see.

**Pull request merged**